File: examples/ts-npm/src/errorHandling.ts
// Fehlerbehandlung — unsaubere Behandlung exponiert Interna (CWE-209).

// UNSICHER — gibt error.message/stack an den Client → leakt Interna, "fail open".
export function handleInsecure(err: unknown): string {
  const e = err as Error;
  return `Fehler: ${e.message}\n${e.stack}`;
}

// SICHER — Detail nur ins Log, generische Meldung nach außen, fail closed.
export function handleSecure(err: unknown): string {
  console.error("Verarbeitung fehlgeschlagen", err); // Stacktrace nur intern
  return "Anfrage konnte nicht verarbeitet werden"; // generisch für den Client
}

File: examples/java-maven/src/main/java/biz/datainmotion/training/ErrorHandling.java
package biz.datainmotion.training;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.logging.Logger;

/**
 * Fehlerbehandlung — unsaubere Behandlung exponiert Interna (CWE-209:
 * Error Message Containing Sensitive Information).
 */
public class ErrorHandling {

    private static final Logger LOG = Logger.getLogger(ErrorHandling.class.getName());

    /**
     * UNSICHER — leakt Dateipfad, Klassennamen, ggf. Systemdetails an den Aufrufer
     * und „macht weiter" (fail open).
     */
    @Deprecated
    public String readInsecure(String path) {
        try {
            return Files.readString(Path.of(path));
        } catch (Exception e) {
            return "Fehler: " + e + " bei " + Path.of(path).toAbsolutePath();
        }
    }

    /**
     * SICHER — Detail nur intern loggen, generische Meldung nach außen,
     * im Fehlerfall sicher abbrechen (fail closed).
     */
    public String read(String path) {
        try {
            return Files.readString(Path.of(path));
        } catch (IOException e) {
            LOG.warning("Lesen fehlgeschlagen"); // ohne Pfad/PII; Stacktrace nur intern
            throw new IllegalStateException("Anfrage konnte nicht verarbeitet werden");
        }
    }
}
